_data-prepper/pipelines/configuration/processors/translate.md

```diff
@@ -0,0 +1,171 @@
+                map:
+                  404: "Not Found"
+  sink:
+    - stdout:
+```
+
+Then create the following file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with the path of a file containing this json data:
+```json
+{ "status": "404" }
+```
+
+The `translate` processor configuration in `pipeline.yaml` retrieves the `source` value from the event data and compares it against the keys specified under the `targets`. 
+When a match is found, the processor places the corresponding mapped value into the `target` key provided in the configuration.
+
+When you run Data Prepper with the previous `pipeline.yaml`, you should see the following output:
+```json
+{
+  "status": "404",
+  "translated_result": "Not Found"
+}
+```
+
+## More options
+
+The following example shows a more involved mapping with additional configurations for `translate` processor:
+
+```yaml
+processor:
+  - translate:
+      mappings:
+        - source: "status"
+          targets:
+            - target: "translated_result"
+              map:
+                404: "Not Found"
+              default: "default"
+              type: "string"
+              translate_when: "/response != null"
+            - target: "another_translated_result"
+              regex:
+                exact: false
+                patterns:
+                  "2[0-9]{2}" : "Success" # Matches ranges from 200-299
+                  "5[0-9]{2}": "Error"    # Matches ranges form 500-599
+      file: 
+        name: "path/to/file.yaml"
+        aws:
+          bucket: my_bucket
+          region: us-east-1
+          sts_role_arn: arn:aws:iam::123456789012:role/MyS3Role
+```
+
+On the top level, specify `mappings` for inline mapping configuration or `file` for a file that contains mapping configurations. Both `mappings` and `file` options can be specified together, and the processor considers the mappings from both sources for translations. In instances where the pipeline configuration and file mappings share duplicate `source` and `target` pairs, the mappings specified within the pipeline configuration take precedence.
+
+
+## Configuration
+
+You can use the following options to configure the `translate` processor.
+
+| Parameter | Required | Type | Description |
+| :--- | :---  | :--- | :--- |
+| mappings | No | List | Defines inline mappings. For more information, see [mappings](#mappings). |
+| file | No | Map | Points to the file that contains mapping configurations. For more information, see [file](#file). |
+
+### mappings
+
+Each item in the `mappings` configuration contains the following options:
+
+| Parameter | Required | Type | Description |
+| :--- | :--- | :--- | :--- |
+| source | Yes | String or List | The source field to translate. Can be a string or a list of strings. |
+| targets | Yes | List | The target fields configurations, which can include the target field key, translation maps, and more. |
```

Comment:
"fields" should either be "field" or "field's". Also, I'd prefer that we not use "and more" (too ambiguous). Can we use "and other [plural noun]"?